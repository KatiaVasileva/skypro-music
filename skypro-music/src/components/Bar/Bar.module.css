.bar {
    position: absolute;
    bottom: 0;
    left: 0;
    width: 100%;
    background: rgba(28, 28, 28, 0.5);
}

.content {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
    -ms-flex-direction: column;
    flex-direction: column;
}

.playerProgress {
    width: 100%;
    height: 5px;
    background: #2e2e2e;
}

.playerBlock {
    height: 73px;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: horizontal;
    -webkit-box-direction: normal;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: justify;
    -ms-flex-pack: justify;
    justify-content: space-between;
}

.player {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: horizontal;
    -webkit-box-direction: normal;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: start;
    -ms-flex-pack: start;
    justify-content: flex-start;
}

.controls {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: horizontal;
    -webkit-box-direction: normal;
    -ms-flex-direction: row;
    flex-direction: row;
    padding: 0 27px 0 31px;
}

.buttonPrev, .buttonPlay, .buttonNext, .buttonRepeat, .buttonShuffle {
    padding: 5px;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
}

.buttonPrev {
    margin-right: 23px;
}

.buttonPrevSvg {
    width: 15px;
    height: 14px;
}

.buttonPlay {
    margin-right: 23px;
    cursor: pointer;
}

.buttonPlaySvg {
    width: 22px;
    height: 20px;
    fill: #d9d9d9;
}

.buttonNext {
    margin-right: 28px;
    fill: #a53939;
}

.buttonNextSvg {
    width: 15px;
    height: 14px;
    fill: inherit;
    stroke: #d9d9d9;
}