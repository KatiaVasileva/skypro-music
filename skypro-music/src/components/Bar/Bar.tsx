"use client"

import { useTrackContext } from "@/hooks/useTrackContext";
import Player from "../Player/Player";
import { TrackContextType } from "@/context/TrackContext";

function Bar() {
  const { currentTrack } = useTrackContext() as TrackContextType;

  return <>{currentTrack && <Player />}</>;
}

export default Bar;
