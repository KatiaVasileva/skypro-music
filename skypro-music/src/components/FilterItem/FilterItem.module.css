.popup {
    position: relative;
}

.popupContainer {
    position: absolute;
    width: 248px;
    max-height: 305px;
    background-color: #313131;
    border-radius: 12px;
    display: flex;
    margin-top: -40px;
    z-index: 1;
}

.popupContainerAuthor {
    margin-left: 100px;
}

.popupContainerYear {
    margin-left: 262px;
    height: 196px;
    width: 221px;
}

.popupContainerGenre {
    margin-left: 425px;
}

.popupBox {
    width: 180px;
    max-height: 237px;
    display: flex;
    margin-top: 20px;
    overflow-y: hidden;
    overflow-y: auto;
    margin: 34px 34px;
    scrollbar-color: #ffffff #4B4949;
    scrollbar-width: thin;
    min-height: 128px;
    min-width: 153px;
}

.popupContent {
    width: 152px;
    max-height: 232px;
    display: flex;
    flex-direction: column;
    gap: 24px;
    min-height: 128px;
}


.popupText {
    font-size: 18px;
    font-weight: 400;
    line-height: 24px;

    &:hover {
        color: #B672FF;
        text-decoration: underline;
        cursor: pointer;
    }
}