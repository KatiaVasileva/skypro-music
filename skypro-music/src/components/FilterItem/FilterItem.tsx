import styles from "./FilterItem.module.css";
import classNames from "classnames";

type FilterItemProps = {
  filterName: string;
  filterContent: Array<string>;
  selectedValues: Array<string>;
  handleFilterItemClick: (filterElement: string) => void;
  handleFilterElementClick?: () => void;
  isFilterElementClicked?: boolean;

};

function FilterItem({
  filterName,
  filterContent,
  selectedValues,
  handleFilterItemClick,
  handleFilterElementClick,
  isFilterElementClicked,
}: FilterItemProps) {

  return (
    <div className={styles.popup}>
      <div
        className={classNames(
          styles.popupContainer,
          filterName === "performer"
            ? styles.popupContainerAuthor
            : filterName === "year"
            ? styles.popupContainerYear
            : styles.popupContainerGenre
        )}
      >
        <div className={styles.popupBox}>
          <div className={styles.popupContent}>
            {filterContent.map((filterElement, index) => (
              <p
                className={classNames(
                  styles.popupText,
                    {
                      [styles.popupTextActive]:
                        selectedValues.includes(filterElement),
                      [styles.popupTextInactive]:
                        filterName === "year" && 
                        !selectedValues.includes(filterElement) && isFilterElementClicked,
                    },
                )}
                key={index}
                onClick={() => {
                  handleFilterItemClick(filterElement);
                  if (handleFilterElementClick) {
                    handleFilterElementClick();
                  }
                  console.log(isFilterElementClicked);
                }}
              >
                {filterElement}
              </p>
            ))}
          </div>
        </div>
      </div>
    </div>
  );
}

export default FilterItem;
