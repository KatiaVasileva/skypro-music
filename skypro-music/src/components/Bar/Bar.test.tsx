import ReduxProvider from "@/store/ReduxProvider";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import FilterItem from "../FilterItem/FilterItem";
import Bar from "./Bar";

describe("Bar component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <ReduxProvider>
        <Bar />
      </ReduxProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
