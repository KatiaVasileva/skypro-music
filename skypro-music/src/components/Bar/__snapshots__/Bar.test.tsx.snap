// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar component renders correctly 1`] = `<div />`;
